modules/nickname/events/guildMemberUpdate.js

```diff
@@ -0,0 +1,67 @@
+module.exports.run = async function (client, oldGuildMember, newGuildMember) {
+
+    if (!client.botReadyAt) return;
+    if (newGuildMember.guild.id !== client.guild.id) return;
+    if (newGuildMember.nickname === oldGuildMember.nickname && newGuildMember.roles.highest.position === oldGuildMember.guild.me.roles.highest.position) return;
+
+    const roles = client.configurations['nickname']['config'];
+    const moduleModel = client.models['nickname']['User'];
+
+    let rolePrefix = "";
+    let hoistrole
+    if (newGuildMember.roles.hoist === null) {
+        hoistrole = null;
+    } else {
+        hoistrole = newGuildMember.roles.hoist.id;
+    }
+
+    for (const role of roles) {
+        if (role.roleID === hoistrole) {
+            rolePrefix = role.prefix;
+        }
+    }
```

Comment:
why is this a loop?

```suggestion
   rolePrefix = roles.find(r => r.roleID === hoistrole)?.prefix; 
```

The `?` ensures that a possible null value returned by the Array.find function does not trigger a runtime error when trying to access the prefix property. This is called [Optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining), and I'm pointing it out because I haven't been using this a lot a few months ago, but recently came across it again, and it's very handy (:

Reply:
If it isn't it puts "undefined" in the rolePrefix String instead of leaving it how it was before (blank from initiation, so the user doesn't get a prefix at all)

Reply:
Thanks for getting back to me. This is the case of poor type usage, but can be easily addressed by falling back to an empty string when no result has been found. I've updated by previous suggestion.